addons/hr_recruitment/tests/test_recruitment.py

```diff
@@ -113,3 +113,22 @@ def test_application_no_partner_duplicate(self):
         self.env['hr.applicant'].create(applicant_data)
         partner_count = self.env['res.partner'].search_count([('email', '=', 'test@thisisatest.com')])
         self.assertEqual(partner_count, 1)
+
+    def test_target_on_application_hiring(self):
+        """
+        Test that the target is updated when hiring an applicant
+        """
+        job = self.env['hr.job'].create({
+            'name': 'Test Job',
+            'no_of_recruitment': 1,
+        })
+        applicant = self.env['hr.applicant'].create({
+            'candidate_id': self.env['hr.candidate'].create({'partner_name': 'Test Applicant'}).id,
+            'job_id': job.id,
+        })
+        self.assertEqual(job.no_of_recruitment, 1)
+        applicant.stage_id = self.env.ref('hr_recruitment.stage_job5')  # Hired
```

Comment:
To make the test more robust, I'd create two stages and use those instead, so that we don't rely on the data and its eventual changes